boot/systems.go

```diff
@@ -86,14 +97,22 @@ func ClearTryRecoverySystem(dev Device, systemLabel string) error {
 }
 
 // SetTryRecoverySystem sets up the boot environment for trying out a recovery
-// system with given label and adds the new system to the list of current
-// recovery systems in the modeenv. Once done, the caller should request
-// switching to the given recovery system.
+// system with given label in the context of the provided device. The call adds
+// the new system to the list of current recovery systems in the modeenv, and
+// optionally sets a try model, if the device mode is different from the current
+// one, which typically can happen during a remodel. Once done, the caller
+// should request switching to the given recovery system.
 func SetTryRecoverySystem(dev Device, systemLabel string) (err error) {
 	if !dev.HasModeenv() {
 		return fmt.Errorf("internal error: recovery systems can only be used on UC20")
 	}
 
+	// either the current device context, in which case the model will match
+	// the current model in the modeenv, or a remodel device context
+	// carrying a new model, for which we may need to set the try model in
+	// the modeenv
+	model := dev.Model()
```

Comment:
(nitpick^2) It seems we `model` only a few lines down, should we move it closer to where it is used (line 137 it seems)? Or are there some side-effects that I'm not aware of?

Reply:
Will tweak this if I end up pushing more things to this PR. Otherwise I'll do a followup

Reply:
Pushed it here.